SemrelEnablement.md

```diff
@@ -0,0 +1,373 @@
+# Enabling semantic-release within Travis builds
+This document provides information on how to enable the use of the `semantic-release` tool
+to automate the publishing of new releases of your Go, Java, Node.js and Python SDK projects.
+
+## Table of Contents
+<!--
+  The TOC below is generated using the `markdown-toc` node package.
+
+      https://github.com/jonschlinkert/markdown-toc
+
+  You should regenerate the TOC after making changes to this file.
+
+      markdown-toc -i --maxdepth 4 SemrelEnablement.md
+  -->
+
+<!-- toc -->
+
+- [Overview](#overview)
+  * [What is semantic-release?](#what-is-semantic-release)
+  * [Assumptions](#assumptions)
+- [Configuring semantic-release](#configuring-semantic-release)
+  * [1. Clone your SDK project](#1-clone-your-sdk-project)
+  * [2. Establish baseline release](#2-establish-baseline-release)
+    + [i. Prepare the baseline release](#i-prepare-the-baseline-release)
+    + [ii. Create initial git tag](#ii-create-initial-git-tag)
+  * [3. Set credentials in Travis build settings](#3-set-credentials-in-travis-build-settings)
+    + [All SDK projects](#all-sdk-projects)
+    + [Go SDK projects](#go-sdk-projects)
+    + [Java SDK projects](#java-sdk-projects)
+    + [Node.js SDK projects](#nodejs-sdk-projects)
+    + [Python SDK projects](#python-sdk-projects)
+  * [4. Enable deploy stage within .travis.yml](#4-enable-deploy-stage-within-travisyml)
+- [Now what?](#now-what)
+- [References:](#references)
+
+<!-- tocstop -->
+
+## Overview
+This document will assist you in configuring your Go, Java, Node.js, and Python SDK projects 
+to use the [semantic-release](https://semantic-release.gitbook.io/semantic-release/)
+tool to automatically publish new releases.
+
+### What is semantic-release?
+`semantic-release` is a tool used during your SDK project's automated CI/CD builds to 
+automate the creation and publishing of new releases of your project.
+`semantic-release` does this by analyzing the commit messages associated with the commits
+that have been merged into the project's `main` branch since the last release was published.
+After analyzing the commit messages, `semantic-release` determines the next
+[semantic version](https://semver.org/) number, produces a changelog and publishes the release,
+including pushing commits and a tag back to your SDK project's git repository.
+
+### Assumptions
+This document assumes the following are true before you try to enable `semantic-release` in your
+SDK project:
+
+1. You have created your SDK project from one of the following SDK template repositories:
+* https://github.ibm.com/CloudEngineering/go-sdk-template
+* https://github.ibm.com/CloudEngineering/java-sdk-template
+* https://github.ibm.com/CloudEngineering/node-sdk-template
+* https://github.ibm.com/CloudEngineering/python-sdk-template
+
+2. You have completed the initial project preparation steps outlined in the SDK template repository's
```

Comment:
I initially had "you've added one or more services" as an assumption, but after thinking about it some more decided that adding service(s) to the project would probably be something one would want to do AFTER the initial baseline release so that those commits show up in the generated changelog, etc.  I sort of settled on the idea that the initial project preparation steps would essentially be the initial baseline release, then after semantic-release is enabled one would add the service(s) and those would cause the version # to be bumped and the changelog to be generated, etc.
What do you think?

Reply:
Ah, that makes sense to me! I forgot too that the prepare project step deletes all of the placeholder example code. I was thinking it would be un-great to include that in the release 🙂 but that's not actually a concern 👍